optimum/amd/ryzenai/configuration.py

```diff
@@ -20,82 +265,316 @@ class QuantizationConfig:
+            If specified, only the nodes in this list are quantized.
+        nodes_to_exclude (List[str], defaults to an empty list `[]`):
+            If specified, nodes in this list will be excluded from quantization.
+        optimize_model (bool, defaults to `True`):
+            If True, optimizes the model before quantization.
+        use_external_data_format (bool, defaults to `False`):
+            Flag for large size (>2GB) models. If True, model proto and data will be stored in separate files.
+        execution_providers (List[str], defaults to `['CPUExecutionProvider']`):
+            Defines the execution providers used by ONNX Runtime for model calibration.
+        convert_fp16_to_fp32 (bool, defaults to `False`):
+            Controls whether to convert the input model from float16 to float32 before quantization.
+        convert_nchw_to_nhwc (bool, defaults to `False`):
+            Controls whether to convert the input NCHW model to NHWC model before quantization.
+        include_cle (bool, defaults to `False`):
+            Flag to optimize models using CrossLayerEqualization; can improve accuracy for some models.
+        extra_options (Union[Dict, None, ExtraOptions], defaults to an instance of `ExtraOptions` with default values):
+            Contains key-value pairs for various options in different cases.
     """
 
-    format: QuantFormat = QuantFormat.QDQ
-    calibration_method: CalibrationMethod = vai_q_onnx.PowerOfTwoMethod.MinMSE
-    activations_dtype: QuantType = QuantType.QUInt8
-    activations_symmetric: bool = True
-    weights_dtype: QuantType = QuantType.QInt8
-    weights_symmetric: bool = True
-    enable_dpu: bool = True
+    format: Literal["qdq", "qop", "vitisqdq"] = "qdq"
+    calibration_method: Literal["nonoverflow", "mse", "minmax", "entropy", "percentile"] = "mse"
+    input_nodes: List[str] = field(default_factory=list)
+    output_nodes: List[str] = field(default_factory=list)
+    op_types_to_quantize: List[str] = field(default_factory=list)
+    random_data_reader_input_shape: Union[List[int], Tuple[int], Dict[str, List[int]]] = field(default_factory=list)
+    per_channel: bool = False
+    reduce_range: bool = False
+    activations_dtype: Literal["uint8", "int8", "uint16", "int16", "uint32", "int32", "bfloat16", "float16"] = "uint8"
+    weights_dtype: Literal["uint8", "int8", "uint16", "int16", "uint32", "int32", "bfloat16", "float16"] = "int8"
+    nodes_to_quantize: List[str] = field(default_factory=list)
+    nodes_to_exclude: List[str] = field(default_factory=list)
+    optimize_model: bool = True
+    use_external_data_format: bool = False
+    execution_providers: List[str] = field(default_factory=lambda: ["CPUExecutionProvider"])
+    enable_ipu_cnn: bool = False
+    convert_fp16_to_fp32: bool = False
+    convert_nchw_to_nhwc: bool = False
+    include_cle: bool = False
+    extra_options: ExtraOptions = field(default_factory=ExtraOptions)
```

Comment:
I don't understand the need for the ExtraOptions class. Why some parameters are there, but `convert_fp16_to_fp32` is not for example? Why not everything at top level?